c++/src/kj/compat/http.c++

```diff
@@ -3490,7 +3549,287 @@ WebSocketPipe newWebSocketPipe() {
 // =======================================================================================
 
 namespace {
+  class WebSocketCompressionUtil {
```

Comment:
I'm considering moving/splitting this code in the following way:
1. Move `CompressionParameters` struct out of `WebSocketCompressionUtil` into `http.h` as a standalone struct. This would mean it could be added as a parameter to `newWebSocket()` in `http.h`.
2. Moving all the static methods out of the class but keeping them in an anonymous namespace. I don't really see a reason to have a class anymore. Previously, I was trying to limit access to the methods to `HttpServer::Connection` and `HttpClientImpl`, but it feels like keeping this within file scope is good enough.

Reply:
Fixup https://github.com/capnproto/capnproto/pull/1475/commits/c9860e420142adb8b0cef6eda5996a47d670f8a2. I figure it'd be nice to change the function/struct names a bit to associate them to websocket stuff, assuming this is a better direction.